simulations/vip-106/simulations.ts

```diff
@@ -0,0 +1,184 @@
+    price: "91.4",
+  },
+  {
+    name: "vXRP",
+    address: "0xB248a295732e0225acd3337607cc01068e3b9c10",
+    price: "0.50599826",
+  },
+  {
+    name: "vBCH",
+    address: "0x5F0388EBc2B94FA8E123F404b79cCF5f40b29176",
+    price: "127.09374798",
+  },
+  {
+    name: "vDOT",
+    address: "0x1610bc33319e9398de5f57b33a5b184c806ad217",
+    price: "6.3763",
+  },
+  {
+    name: "vLINK",
+    address: "0x650b940a1033B8A1b1873f78730FcFC73ec11f1f",
+    price: "7.288",
+  },
+  {
+    name: "vDAI",
+    address: "0x334b3ecb4dca3593bccc3c7ebd1a1c1d1780fbf1",
+    price: "0.99987",
+  },
+  {
+    name: "vFIL",
+    address: "0xf91d58b5aE142DAcC749f58A49FCBac340Cb0343",
+    price: "5.6139",
+  },
+  {
+    name: "vBETH",
+    address: "0x972207A639CC1B374B893cc33Fa251b55CEB7c07",
+    price: "1855.657809",
+  },
+  {
+    name: "vADA",
+    address: "0x9A0AF7FDb2065Ce470D72664DE73cAE409dA28Ec",
+    price: "0.38441773",
+  },
+  {
+    name: "vDOGE",
+    address: "0xec3422ef92b2fb59e84c8b02ba73f1fe84ed8d71",
+    price: "902379100",
+  },
+  {
+    name: "vMATIC",
+    address: "0x5c9476FcD6a4F9a3654139721c949c2233bBbBc8",
+    price: "1.121",
+  },
+  {
+    name: "vCAKE",
+    address: "0x86aC3974e2BD0d60825230fa6F355fF11409df5c",
+    price: "3.66796517",
+  },
+  {
+    name: "vAAVE",
+    address: "0x26DA28954763B92139ED49283625ceCAf52C6f94",
+    price: "79.25683356",
+  },
+  {
+    name: "vTUSD",
+    address: "0x08CEB3F4a7ed3500cA0982bcd0FC7816688084c3",
+    price: "1.00032432",
+  },
+  {
+    name: "vTRX",
+    address: "0x61eDcFe8Dd6bA3c891CB9bEc2dc7657B3B422E93",
+    price: "0.06646645",
+  },
+  {
+    name: "vTRX",
+    address: "0xC5D3466aA484B040eE977073fcF337f2c00071c1",
+    price: "66466450000",
+  },
+  {
+    name: "VAI",
+    address: "0x4BD17003473389A42DAF6a0a729f6Fdb328BbBd7",
+    price: "0.96847512",
+  },
+];
+
+forking(27116217, () => {
+  let resilientOracle: ethers.Contract;
+  let priceOracle: ethers.Contract;
+  const provider = ethers.provider;
+
+  before(async () => {
+    resilientOracle = new ethers.Contract(RESILIENT_ORACLE, RESILIENT_ORACLE_ABI, provider);
+    priceOracle = new ethers.Contract(PRICE_ORACLE, PRICE_ORACLE_ABI, provider);
+    await setMaxStalePeriodInOracle(COMPTROLLER);
+  });
+
+  describe("Pre-VIP behavior", async () => {
+    it("validate vToken prices", async () => {
+      for (let i = 0; i < vTokens.length; i++) {
+        const vToken = vTokens[i];
+        const price = await priceOracle.getUnderlyingPrice(vToken.address);
+        expect(price).to.be.equal(parseUnits(vToken.price, 18));
+      }
+    });
+  });
+
+  testVip("VIP-106 Change Oracle and Configure Resilient Oracle", vip106());
+
+  describe("Post-VIP behavior", async () => {
+    it("validate vToken prices", async () => {
+      for (let i = 0; i < vTokens.length; i++) {
+        const vToken = vTokens[i];
+        const price = await resilientOracle.getUnderlyingPrice(vToken.address);
```

Comment:
We can check also the oracle set in the comptroller contract (that should be changed too), and also the prices returned by the oracle configured in the comptroller (not directly using the resilientOracle reference)